core/controllers/contributor_dashboard.py

```diff
@@ -308,25 +338,32 @@ def get(self):
 
         exp = exp_fetchers.get_exploration_by_id(exp_id, strict=False)
         if exp is None:
-            raise self.PageNotFoundException('Invalid exp_id: %s' % exp_id)
+            raise self.PageNotFoundException()
         state_names_to_content_id_mapping = exp.get_translatable_text(
             target_language_code)
-        if not state_names_to_content_id_mapping.has_key(state_name):
-            raise self.PageNotFoundException(
-                'Invalid state_name: %s' % state_name)
-        content_id_mapping = state_names_to_content_id_mapping[state_name]
+        if state_name not in state_names_to_content_id_mapping:
+            raise self.PageNotFoundException()
+        content_id_to_text_mapping = (
+            state_names_to_content_id_mapping[state_name])
         translated_texts = {}
+        errors = {}
         for content_id in content_ids:
-            if content_id_mapping.has_key(content_id):
-                source_text = content_id_mapping[content_id]
+            try:
```

Comment:
The part in the "try" part of a try/except should be as small as possible. Which part exactly are you thinking might fail here?

Note that "extra" content IDs are just a regular 400 error. It means that the client has sent bad input.

Reply:
I was thinking that `translation_services.get_and_cache_machine_translation` might error, but I forgot I set it up to return None for an unsuccessful translation. I'll also return None for an "extra" content ID.

Reply:
I removed the try catch, this now returns None for failures to translate specific content_ids with no error info sent to the front end.